pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -289,42 +298,57 @@ func decodeIPSecKey(keyRaw string) ([]byte, error) {
 
 // LoadIPSecKeysFile imports IPSec auth and crypt keys from a file. The format
 // is to put a key per line as follows, (auth-algo auth-key enc-algo enc-key)
-func LoadIPSecKeysFile(path string) error {
+func LoadIPSecKeysFile(path string) (uint8, error) {
 	file, err := os.Open(path)
 	if err != nil {
-		return err
+		return 0, err
 	}
 	defer file.Close()
 	return loadIPSecKeys(file)
 }
 
-func loadIPSecKeys(r io.Reader) error {
+func loadIPSecKeys(r io.Reader) (uint8, error) {
+	var spi uint8
+
 	scanner := bufio.NewScanner(r)
 	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
+		offset := 0
+
 		ipSecKey := &ipSecKey{
-			Spi:   1,
 			ReqID: 1,
 		}
 
 		// Scanning IPsec keys formatted as follows,
 		//    auth-algo auth-key enc-algo enc-key
 		s := strings.Split(scanner.Text(), " ")
 		if len(s) < 4 {
-			return fmt.Errorf("missing IPSec keys or invalid format")
+			return 0, fmt.Errorf("missing IPSec keys or invalid format")
 		}
 
-		authkey, err := decodeIPSecKey(s[1])
+		spiI, err := strconv.Atoi(s[0])
```

Comment:
What is spiI? ID, right?

Reply:
Here spi is security parameter index and I fell into hungarian naming convention it seems just to indicate that spiI is the integer version here giving `spiI`. We do need to be a bit careful here because the spi is shifted into the mark field and if we shift a uint8 << 12 well there is nothing left by C standards.